internal/mesh/internal/types/intermediate/types.go

```diff
@@ -27,41 +28,47 @@ type CombinedDestinationRef struct {
 	ExplicitDestinationsID *pbresource.ID
 }
 
+// Deprecated: types.DecodedServiceEndpoints
 type ServiceEndpoints struct {
 	Resource  *pbresource.Resource
 	Endpoints *pbcatalog.ServiceEndpoints
 }
 
+// Deprecated: types.DecodedService
 type Service struct {
 	Resource *pbresource.Resource
 	Service  *pbcatalog.Service
 }
 
+// Deprecated: types.DecodedDestinations
 type Destinations struct {
 	Resource     *pbresource.Resource
 	Destinations *pbmesh.Upstreams
 }
 
+// Deprecated: types.DecodedWorkload
 type Workload struct {
 	Resource *pbresource.Resource
 	Workload *pbcatalog.Workload
 }
 
+// Deprecated: types.DecodedProxyStateTemplate
 type ProxyStateTemplate struct {
 	Resource *pbresource.Resource
 	Tmpl     *pbmesh.ProxyStateTemplate
 }
 
+// Deprecated: types.DecodedProxyConfiguration
```

Comment:
I think we can remove these as there are no other PRs stacked on top currently (or at least nothing big)